joomla/joomla-cms
# mySQL 5.7 nullDate compatibility fix

Summary of Changes

implemented the nullDate change for mysql 5.7 (from 0000-00-00 00:00:00 to 1000-01-01 00:00:00)
Testing Instructions

installation and/or updating on mysql 5.6 server, all datetime database default values of joomla should be 0000-00-00 00:00:00,

on installation and/or updating on mysql 5.7 server, all datetime database default values of joomla should be 1000-01-01 00:00:00

this could break external components, who rely on hardcoded nullDates.

it also implements #11527

the update routine should work in both ways (upgrade and downgrade of mysql version) - downgrade is experimental.

TESTING:
you need at least 2 different test enviroments and a working fake upgrade server
it has to be test with:
new install with mysql <=5.6
new install with mysql >=5.7
upgrade from <=5.6 to 5.7
downgrade from 5.7 to <=5.6 (optional)

be aware, that some precompiled binaries of mysql 5.7 accept the old null values, so it could be required to compile it from scratch with the correct flags and configuration.

every update step has to been done multiple times using:
Update using FTP upload and FIX Database button (Extensions->Manage->Database)
Using the auto-updater module (Joomla:Update)
Using the manual update uploader

It's working if every nullDate value in the database is either 0000-00-00 or 1000-01-01 (depending on mysql 5.6 / 5.7)

this patch has been created at @icampus
